param/__init__.py

```diff
@@ -775,22 +773,37 @@ def __init__(self,default=(0,0),length=None,**params):
             self.length = length
 
         self._check(default)
-        Parameter.__init__(self,default=default,**params)
+        super(Tuple,self).__init__(default=default,**params)
+
 
     def _check(self,val):
+        if val is None and self.allow_None:
+            return
+
         if not isinstance(val,tuple):
             raise ValueError("NumericTuple '%s' only takes a tuple value."%self._attrib_name)
 
         if not len(val)==self.length:
             raise ValueError("%s: tuple is not of the correct length (%d instead of %d)." %
                              (self._attrib_name,len(val),self.length))
-        for n in val:
-            if not _is_number(n):
-                raise ValueError("%s: tuple element is not numeric: %s." % (self._attrib_name,str(n)))
+
 
     def __set__(self,obj,val):
         self._check(val)
-        super(NumericTuple,self).__set__(obj,val)
+        super(Tuple,self).__set__(obj,val)
+
+
+
+class NumericTuple(Tuple):
+    """A numeric tuple Parameter (e.g. (4.5,7.6,3)) with a fixed tuple length."""
+
+    __slots__ = []
```

Comment:
So what should we do going forward? Should we simply rely on the metaclass to patch the Parameters with __slots__ or be explicit and declare "empty" slots? The documentation is there now but it's still non-obvious and I probably would have missed it had I not gone out of my way to learn about slots.